cub/test/catch2_test_warp_scan_api.cu

```diff
@@ -0,0 +1,173 @@
+/******************************************************************************
+ * Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the NVIDIA CORPORATION nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NVIDIA CORPORATION BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ ******************************************************************************/
+
+#include <cub/warp/warp_scan.cuh>
+
+#include <thrust/detail/raw_pointer_cast.h>
+#include <thrust/device_vector.h>
+#include <thrust/host_vector.h>
+
+#include "catch2_test_helper.h"
+#include "cuda/std/__algorithm/fill.h"
+#include "cuda/std/__algorithm/max.h"
+#include "cuda/std/__numeric/inclusive_scan.h"
+#include "cuda/std/__numeric/iota.h"
+
+constexpr int num_warps = 4;
+
+template <typename T>
+struct max_op
+{
+  __host__ __device__ T const& operator()(T const& i, T const& j)
+  {
+    return cuda::std::max(i, j);
+  }
+};
+
+template <typename T>
+struct sum_op
+{
+  __host__ __device__ T operator()(T const& i, T const& j)
+  {
+    return i + j;
+  }
+};
+
+// example-begin inclusive-warp-scan-init-value
+__global__ void InclusiveScanKernel(int* output)
+{
+  // Specialize WarpScan for type int
+  typedef cub::WarpScan<int> warp_scan_t;
```

Comment:
suggestion: we tend to use using aliases when possible. Those are easier to read because they follow the usual order of "name = definition" a opposed to the reversed order in typedefs "definition name":
```suggestion
  using warp_scan_t = cub::WarpScan<int>;
```